// react-css-dropdown/dist/index.modern.js
import React from 'react';
import clsx from 'clsx';

var styles = {"selectContainer":"_3Ifyx","select":"_1ul-b","option":"_3tQrm","arrow":"_35l6Q"};

var Dropdown = function Dropdown(_ref) {
  var options = _ref.options,
      _ref$defaultSelection = _ref.defaultSelection,
      defaultSelection = _ref$defaultSelection === void 0 ? {
    name: 'Choose selection',
    isActive: true
  } : _ref$defaultSelection,
      handleSelect = _ref.handleSelect,
      containerClass = _ref.containerClass,
      selectClass = _ref.selectClass,
      arrowClass = _ref.arrowClass;
  return React.createElement("div", {
    className: clsx(styles.selectContainer, containerClass)
  }, React.createElement("select", {
    className: clsx(styles.select, selectClass),
    defaultValue: defaultSelection.name,
    onChange: handleSelect
  }, defaultSelection.isActive && React.createElement("option", {
    disabled: true,
    value: defaultSelection.name
  }, defaultSelection.name), options.map(function (_ref2, index) {
    var name = _ref2.name,
        value = _ref2.value,
        _ref2$disabled = _ref2.disabled,
        disabled = _ref2$disabled === void 0 ? false : _ref2$disabled;
    return React.createElement("option", {
      key: index,
      value: value,
      disabled: disabled
    }, name);
  })), React.createElement("span", {
    className: clsx(styles.arrow, arrowClass),
    tabIndex: 0
  }));
};

export { Dropdown };
